src/Microsoft.AspNet.Mvc.Core/DefaultControllerTypeProvider.cs

```diff
@@ -76,17 +77,45 @@ protected internal virtual bool IsController([NotNull] TypeInfo typeInfo)
             {
                 return false;
             }
-            if (typeInfo.Name.Equals("Controller", StringComparison.OrdinalIgnoreCase))
+            if (typeInfo.Name.Equals(ControllerTypeName, StringComparison.OrdinalIgnoreCase))
```

Comment:
This is to ensure we don't capture our own controller, however what if a user has defined a `Controller` class for something completely non-mvc specific? I know you didn't add this piece of code but we should probably be validating the `FullName` here vs the type of `Controller`